plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/TokenFetchingMetastoreClientFactory.java

```diff
@@ -106,12 +119,22 @@ private DelegationToken getDelegationToken(String username)
 
     private DelegationToken loadDelegationToken(String username)
     {
-        try (ThriftMetastoreClient client = createMetastoreClient()) {
-            return new DelegationToken(System.nanoTime(), client.getDelegationToken(username));
+        try {
+            // added retry and stats for the thrift delegation token
+            return retry()
```

Comment:
We're trying to deprecate `RetryDriver`. Please use `Failsafe` instead.

Reply:
updated, thanks